### PR TITLE
chore: bump rust version to latest nightly

### LINKED ISSUE #3452: Remove false positive clippy lint on next rust update
### File Location(s)

`examples/constructor/src/builder.rs`
`scripts/src/clippy.sh`
rust-lang/rust-clippy#11545

### Proposal

```rust
    // TODO #3452: remove this on next rust update
    #[allow(clippy::useless_conversion)]
    pub fn add_from_iter(mut self, calls: impl Iterator<Item = Call>) -> Self { ... }
```

### PR DESCRIPTION
**Pull request by StackOverflowExcept1on**
Resolves #3452

TODO:
- [x] check some stuff related to clippy
- [x] find more stable version

### CODE REVIEW

File: common/numerated/src/interval.rs [hunk 1]
```diff
@@ -441,7 +441,7 @@ mod tests {
     #[test]
     fn size() {
         assert_eq!(Interval::<u8>::try_from(11..111).unwrap().size(), Some(100),);
-        assert_eq!(Interval::<u8>::try_from(..1).unwrap().size(), Some(1),);
+        assert_eq!(Interval::<u8>::from(..1).size(), Some(1),);
```
Comment (shamilsan) [inline common/numerated/src/interval.rs:444]: What is the difference between `try_from` + `unwrap` and `from`?

**Review by shamilsan — APPROVED** [review]

File: common/numerated/src/interval.rs (hunk 1 above)
Comment (StackOverflowExcept1on) [reply common/numerated/src/interval.rs:444]: clippy reported that this code is infallible and suggested to use `From::from`

**Review by SabaunT — COMMENTED** [review]
Has it fixed all the stuff related to fuzzer and coverage?

**Comment by StackOverflowExcept1on** [comment]
but this should not be merged until rust-lang/rust-clippy#11953

**Comment by breathx** [comment]
Re-open once fuzzer works well

File: gclient/src/api/listener/mod.rs [hunk 2]
```diff
@@ -21,7 +21,6 @@ mod subscription;
 
 pub use gsdk::metadata::{gear::Event as GearEvent, Event};
 pub use iterator::*;
-pub use subscription::*;
```
Comment (breathx) [inline gclient/src/api/listener/mod.rs:24]: why

File: gclient/src/lib.rs [hunk 3]
```diff
@@ -135,7 +135,7 @@ mod api;
 mod utils;
 mod ws;
 
-pub use api::{calls::*, error::*, listener::*, GearApi};
```
Comment (breathx) [inline gclient/src/lib.rs:138]: and here

File: gstd/src/macros/debug.rs [hunk 4]
```diff
@@ -69,14 +69,15 @@ macro_rules! debug {
 #[macro_export]
 macro_rules! dbg {
     () => {
-        $crate::debug!("[{}:{}]", $crate::prelude::file!(), $crate::prelude::line!())
+        $crate::debug!("[{}:{}:{}]", $crate::prelude::file!(), $crate::prelude::line!(), $crate::prelude::column!())
```
Comment (breathx) [inline gstd/src/macros/debug.rs:72]: isn't it excess info?

File: runtime/vara/src/lib.rs [hunk 5]
```diff
@@ -270,8 +270,7 @@ impl pallet_babe::Config for Runtime {
     type WeightInfo = ();
     type MaxAuthorities = MaxAuthorities;
 
-    type KeyOwnerProof =
-        <Historical as KeyOwnerProofSystem<(KeyTypeId, pallet_babe::AuthorityId)>>::Proof;
+    type KeyOwnerProof = sp_session::MembershipProof;
```
Comment (breathx) [inline runtime/vara/src/lib.rs:273]: what if it gonna change in future, it's easy to miss it now

**Review by breathx — APPROVED** [review]

File: gclient/src/api/listener/mod.rs (hunk 2 above)
Comment (StackOverflowExcept1on) [reply gclient/src/api/listener/mod.rs:24]: clippy

File: gstd/src/macros/debug.rs (hunk 4 above)
Comment (StackOverflowExcept1on) [reply gstd/src/macros/debug.rs:72]: no, 1.76 added this https://github.com/rust-lang/rust/pull/114962

File: runtime/vara/src/lib.rs (hunk 5 above)
Comment (StackOverflowExcept1on) [reply runtime/vara/src/lib.rs:273]: just copy this from polkadot rococo-runtime if this has changed
https://github.com/paritytech/polkadot-sdk/pull/2644

File: gstd/src/macros/debug.rs (hunk 4 above)
Comment (breathx) [reply gstd/src/macros/debug.rs:72]: > no, 1.76 added this [rust-lang/rust#114962](https://github.com/rust-lang/rust/pull/114962)

I mean for our case of calling it in gear programs

File: gclient/src/lib.rs (hunk 3 above)
Comment (breathx) [reply gclient/src/lib.rs:138]: checked it, really unnecessary

File: gstd/src/macros/debug.rs (hunk 4 above)
Comment (StackOverflowExcept1on) [reply gstd/src/macros/debug.rs:72]: I can revert that but mb column is useful to jump to source file?

Comment (breathx) [reply gstd/src/macros/debug.rs:72]: not a big deal

**Comment by breathx** [comment]
Btw dont we want just whitelist `#[allow(static_mut_ref)]` ?
We have single thread guarantees here

**Review by techraed — APPROVED** [review]

**Pull request merged**